crates/aptos-indexer-api/src/graphql_root_queries/transaction_queries.rs

```diff
@@ -0,0 +1,53 @@
+use crate::graphql_root_queries::AptosTransaction;
+use aptos_indexer::{
+    database::PgPoolConnection,
+    models::{
+        transactions::{Transaction},
+    },
+};
+use async_graphql::{FieldResult};
+
+pub fn get_transaction_by_version(
+    version: u64,
+    conn: &PgPoolConnection,
+) -> FieldResult<AptosTransaction> {
+    let (txn, maybe_user_txn, maybe_block_metadata_txn, events, writesets) =
+        Transaction::get_by_version(version, conn).unwrap();
+    Ok(AptosTransaction {
```

Comment:
would be better to reconstruct the ful type (and/or mark the empty field (user_tx or block_meta) as don't serialize if option::is_none in serde)